app/public/src/pages/game.tsx

```diff
@@ -292,6 +294,24 @@ export default function Game() {
         gameContainer.onDragDropCombine(event)
       }) as EventListener)
 
+      room.onLeave((code) => {
+        const shouldReconnect = code === CloseCodes.ABNORMAL_CLOSURE || code === CloseCodes.TIMEOUT
+        logger.info(`left game room with code ${code}`, { shouldReconnect })
+        if (shouldReconnect) {
+          logger.log("Connection closed unexpectedly or timed out. Attempting reconnect.")
+          // Clear out all game data to reload the game
+          connected.current = false
+          initialized.current = false
+          if (gameContainer && gameContainer.game) {
+            gameContainer.game.destroy(true)
```

Comment:
i don't see why a socket disconnection should imply destroying and recreating the game canvas. 

The only thing needed to reconnect after a websocket abnormal closure is `connectToGame` and set up the event listeners on the room. Therefore all the `room.onMessage` and `room.state.listen` could be moved to `connectToGame` after room is reconnected, so that the only thing you need to do is `connected.current = false`